src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py

```diff
@@ -286,6 +315,8 @@ def closeBlocks(self):
             myThread.transaction.begin()
             try:
                 xmlData = self.createInjectionSpec(migratedBlocks[siteName])
+                # inject and subscribe when closing the block
+                subRes = self.subscribeBlocksInDataset(migratedBlocks[siteName].keys(), location, xmlData)
```

Comment:
I would not make two remote API calls in the same try/except block. Not without thinking about what would happen if the first call suceeds and the first fails. First here is block subscriptions, is it safe to retry that even if it already succeeded ? Even if it is, two try/except blocks would be better strictly to know in the execption handler which call did in fact fail.